pkg/performanceprofile/controller/performanceprofile_controller_hypershift.go

```diff
@@ -71,3 +102,230 @@ func (r *PerformanceProfileReconciler) hypershiftSetupWithManager(mgr ctrl.Manag
+			return fmt.Errorf("failed to update configmap %q: %w", cm.Name, err)
+		}
+	}
+	return nil
+}
+
+func newOwnerReference(owner metav1.Object, gvk schema.GroupVersionKind) metav1.OwnerReference {
+	blockOwnerDeletion := false
+	isController := false
+	return metav1.OwnerReference{
+		APIVersion:         gvk.GroupVersion().String(),
+		Kind:               gvk.Kind,
+		Name:               owner.GetName(),
+		UID:                owner.GetUID(),
+		BlockOwnerDeletion: &blockOwnerDeletion,
+		Controller:         &isController,
+	}
+}
+
+func TunedConfigMap(owner *corev1.ConfigMap, performanceProfileName, nodePoolNamespacedName, tunedManifest string) *corev1.ConfigMap {
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: owner.GetNamespace(),
+			Name:      "tuned-" + owner.Name,
+			Labels: map[string]string{
+				tunedConfigMapLabel:                   "true",
+				hypershiftPerformanceProfileNameLabel: performanceProfileName,
+				hypershiftNodePoolLabel:               parseNamespacedName(nodePoolNamespacedName),
+			},
+			Annotations: map[string]string{
+				hypershiftNodePoolLabel: nodePoolNamespacedName,
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				newOwnerReference(owner, owner.GroupVersionKind()),
+			},
+		},
+		Data: map[string]string{
+			tunedConfigMapConfigKey: tunedManifest,
+		},
+	}
+}
+
+// parseManifests parses a YAML or JSON document that may contain one or more
+// kubernetes resources.
+func parsePerformanceProfileManifest(data []byte, nodePoolName string) (*performancev2.PerformanceProfile, error) {
+	scheme := runtime.NewScheme()
+	performancev2.AddToScheme(scheme)
+
+	//REVIEW - As serializer is used many times it could be worthy to create it just one time and keep it somewhere
+	yamlSerializer := serializer.NewSerializerWithOptions(
+		serializer.DefaultMetaFactory, scheme, scheme,
+		serializer.SerializerOptions{Yaml: true, Pretty: true, Strict: true},
+	)
+
+	cr, _, err := yamlSerializer.Decode(data, nil, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding PerformanceProfile manifests: %v", err)
+	}
+	performanceProfile, ok := cr.(*performancev2.PerformanceProfile)
+	if !ok {
+		return nil, fmt.Errorf("error parsing PerformanceProfile manifests")
+	}
+
+	// Make PerformanceProfile names unique if a PerformanceProfile is duplicated across NodePools
+	// for example, if one ConfigMap is referenced in multiple NodePools
+	performanceProfile.SetName(performanceProfile.ObjectMeta.Name + "-" + hashStruct(nodePoolName))
+	klog.V(2).Infof("parsePerformanceProfileManifest: name: %s", performanceProfile.GetName())
+
+	// Propagate NodePool name from ConfigMap down to PerformanceProfile object
+	if performanceProfile.Labels == nil {
+		performanceProfile.Labels = make(map[string]string)
+	}
+	performanceProfile.Labels[hypershiftNodePoolNameLabel] = nodePoolName
+
+	return performanceProfile, nil
+}
+
+func hashStruct(o interface{}) string {
```

Comment:
but we use this to hash a string, don't we?

Reply:
yes it is used to hash a string.